scala/lambdas/preingest-tdr-aggregator/src/main/scala/uk/gov/nationalarchives/preingesttdraggregator/Aggregator.scala

```diff
@@ -81,45 +76,45 @@ object Aggregator:
       )
     }
 
-    private def startNewGroup(groupCacheRef: Ref[F, Map[String, Group]], sourceId: String, config: Config, lambdaTimeoutTime: MilliSeconds)(using
+    private def startNewGroup(sourceId: String, config: Config, groupExpiryTime: Long)(using
         sfnClient: DASFNClient[F],
         enc: Encoder[SFNArguments]
-    ): F[GroupId] = {
-      val groupExpiryTime = lambdaTimeoutTime + (config.maxSecondaryBatchingWindow * 1000)
+    ): F[Group] = {
       val waitFor: Seconds = Math.ceil((groupExpiryTime - Generators().generateInstant.toEpochMilli).toDouble / 1000).toInt.seconds
       val groupId: GroupId = GroupId(config.sourceSystem)
       val batchId = BatchId(groupId)
-      for {
-        _ <- sfnClient.startExecution(config.sfnArn, SFNArguments(groupId, batchId, waitFor), batchId.batchValue.some)
-        _ <- groupCacheRef.update(groupCache => groupCache + (sourceId -> Group(groupId, Instant.ofEpochMilli(groupExpiryTime), 1)))
-      } yield groupId
+      sfnClient.startExecution(config.sfnArn, SFNArguments(groupId, batchId, waitFor), batchId.batchValue.some).map { _ =>
+        Group(groupId, Instant.ofEpochMilli(groupExpiryTime), 1)
+      }
     }
 
-    private def getNewOrExistingGroupId(groupCacheRef: Ref[F, Map[String, Group]], config: Config, sourceId: String, lambdaTimeoutTime: MilliSeconds)(using
+    private def getNewOrExistingGroupId(atomicCell: AtomicCell[F, Map[String, Group]], config: Config, sourceId: String, lambdaTimeoutTime: MilliSeconds)(using
         Encoder[SFNArguments]
     ): F[GroupId] = {
       def log = logWithReason(sourceId)
-      groupCacheRef.get.flatMap { groupCache =>
-        val potentialCurrentGroupForSource = groupCache.get(sourceId)
-        potentialCurrentGroupForSource match
-          case None => log(NoExistingGroup) >> startNewGroup(groupCacheRef, sourceId, config, lambdaTimeoutTime)
-          case Some(currentGroupForSource) =>
-            if currentGroupForSource.expires.toEpochMilli <= lambdaTimeoutTime then
-              log(ExpiryBeforeLambdaTimeout) >> startNewGroup(groupCacheRef, sourceId, config, lambdaTimeoutTime)
-            else if currentGroupForSource.itemCount > config.maxBatchSize then log(MaxGroupSizeExceeded) >> startNewGroup(groupCacheRef, sourceId, config, lambdaTimeoutTime)
-            else groupCacheRef.update(_ + (sourceId -> currentGroupForSource.copy(itemCount = currentGroupForSource.itemCount + 1))) >> Async[F].pure(currentGroupForSource.groupId)
-      }
+      val groupExpiryTime = lambdaTimeoutTime + (config.maxSecondaryBatchingWindow * 1000)
```

Comment:
Could this be converted to `Milliseconds`?

Reply:
Yeah, it can. I've changed a couple of bits so we've got Seconds and MilliSeconds in most places.